src/dr_log.cpp

```diff
@@ -268,15 +275,10 @@ void setupLogging(std::string const & log_file, std::string const & name) {
 	log::add_common_attributes();
 	core->add_global_attribute("Node", log::attributes::constant<std::string>(name));
 
-	// Get environment variables to check if we should skip some sinks.
-	char * use_console_sink   = std::getenv("DR_LOG_USE_CONSOLE");
-	char * use_syslog_sink    = std::getenv("DR_LOG_USE_SYSLOG");
-	char * console_format    = std::getenv("DR_LOG_CONSOLE_FORMAT");
-
-	bool console_sink_enabled = ((!use_console_sink || std::atoi(use_console_sink)) && (!console_format || (strcmp("systemd", console_format) != 0)));
-	bool syslog_sink_enabled  = (!use_syslog_sink  || std::atoi(use_syslog_sink));
-	bool systemd_sink_enabled = ((!use_console_sink || std::atoi(use_console_sink)) && (!console_format || (strcmp("systemd", console_format) == 0)));
-
+	// Get environment variables to check which sinks to enable
+	bool console_sink_enabled = ((getEnv("DR_LOG_USE_CONSOLE") != "0") && (getEnv("DR_LOG_CONSOLE_FORMAT") != "systemd"));
+	bool systemd_sink_enabled = ((getEnv("DR_LOG_USE_CONSOLE") != "0") && (getEnv("DR_LOG_CONSOLE_FORMAT") == "systemd"));
+	bool syslog_sink_enabled  = (getEnv("DR_LOG_USE_SYSLOG") != "0");
```

Comment:
It's a tiny bit of a shame to perform the env lookup and string conversion each time. You could consider storing `use_console` and `console_format` in strings once and then checking those.

However, it's not a big problem since this is only done once at startup. So feel free to ignore :)

Reply:
If you decide to ignore this, I think the PR is ready to be merged.

Reply:
We're never getting back the cycles that my IDE spend on this fix. :P - But when you're right you're right. Solved!